beacon-chain/state/types/types.go

```diff
@@ -95,6 +97,170 @@ func (f FieldIndex) String(stateVersion int) string {
 	}
 }
 
+// RealPosition denotes the position of the field in the provided beacon state.
+// The value might differ for different state versions.
+func (f FieldIndex) RealPosition(s beaconState.BeaconState) int {
```

Comment:
This is duplicated in `nativestate`, but it's temporary. Once we get rid of proto state, this will be gone too.